signals/webhook/webhook.go

```diff
@@ -38,27 +37,67 @@ const (
 
 // Note: micro requires stateless operation so the Listen() method should not use the
 // receive struct to save or modify state.
-type webhook struct{}
+// webhooks are special in that they should only have one http Server since the port is fixed at runtime
+// this means that webhook signals are stateful, however losing this state is not a concern since
+// the connections will be re-initialized by the SignalClient
+type webhook struct {
+	srv *http.Server
+	sync.RWMutex
+	inactiveEndpoints map[string]struct{}
```

Comment:
Can we use Golang's sync map instead? https://golang.org/pkg/sync/#Map

Reply:
so use cases are:
- when the entry for a given key is only ever written once but read many times, as in caches that only grow
- when multiple goroutines read, write, and overwrite entries for disjoint sets of keys

I think both of these may apply to some degree.. so it may be worth losing the type safety over this...